### ISSUE TITLE
Enhancement: [keyword-spacing] Make it possible to force a space after `import type`

### ISSUE THREAD

**Issue by ntnyq** [typescript-eslint/typescript-eslint#5362]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/keyword-spacing

### Description

I read ESLint docs about the rule `keyword-spacing` and it supports only [these keywords](https://eslint.org/docs/latest/rules/keyword-spacing#rule-details).

Besides, I think the keyword `type` should be supported by `typescript-eslint` rather than ESLint.

### Fail

```typescript
// with options to force a space after import type
import type{ FooBar } from './types'
```

### Pass

```typescript
import type { FooBar } from './types'
```
